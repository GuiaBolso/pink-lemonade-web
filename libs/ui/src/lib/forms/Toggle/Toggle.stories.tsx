import React from 'react';
import Toggle from './Toggle';

export default {
  title: 'Forms/Toggle',
  component: Toggle.type,
};

export const simple = () => <Toggle name="newsletter" label="Simple" />;

export const checked = () => (
  <Toggle defaultChecked label="Checked" name="newsletter" />
);

export const disabled = () => (
  <Toggle label="Disabled" checked disabled name="newsletter" />
);
