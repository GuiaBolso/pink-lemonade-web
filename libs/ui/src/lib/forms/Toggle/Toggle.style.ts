import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { default as MuiFormControlLabel } from '@material-ui/core/FormControlLabel';
import { CustomThemeProps } from '../../../typings/CustomThemeProps';

type ToggleStyledProps = {
  theme: CustomThemeProps;
};

export const FormControlLabel = styled(MuiFormControlLabel)`
  ${({ theme }: ToggleStyledProps) => {
    return css`
      && {
        .MuiSwitch-root {
          color: ${theme?.colors?.neutral?.dark};

          .Mui-checked {
            color: ${theme?.colors?.brand?.secondary?.default};

            & + .MuiSwitch-track {
              background-color: ${theme?.colors?.brand?.secondary?.default};
            }
          }

          .Mui-disabled {
            color: ${theme?.colors?.neutral?.light};
          }
        }
      }

      .MuiTypography-root {
        font-size: inherit;
        font-family: ${theme?.type?.fontFamily?.neutral};
        font-weight: inherit;
        line-height: inherit;
        letter-spacing: inherit;

        &.Mui-disabled {
          color: ${theme?.colors?.neutral?.light};
        }
      }
    `;
  }};
`;
