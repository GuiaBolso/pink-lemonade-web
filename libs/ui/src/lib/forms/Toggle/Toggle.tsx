import React, { memo } from 'react';
import { FormControlLabelProps, SwitchProps } from '@material-ui/core';

import Switch from '@material-ui/core/Switch';

import * as S from './Toggle.style';

type ToggleProps = {
  name?: string;
  size?: 'small' | 'medium';
} & Partial<FormControlLabelProps>;

const Toggle = ({
  label,
  name,
  disabled = false,
  size = 'medium',
  defaultChecked,
  ...rest
}: ToggleProps) => {
  let toggleProps: SwitchProps = { name };

  defaultChecked &&
    (toggleProps = {
      ...toggleProps,
      defaultChecked,
    });

  return (
    <S.FormControlLabel
      {...rest}
      label={label}
      disabled={disabled}
      control={<Switch {...toggleProps} size={size} />}
    />
  );
};

export default memo(Toggle);
