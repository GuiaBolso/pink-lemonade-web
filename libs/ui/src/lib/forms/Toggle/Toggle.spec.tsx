import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import Toggle from './Toggle';

describe('Toggle Component', () => {
  const makeSetup = () => {
    const element = <Toggle label="Lorem Ipsum" name="simple-toggle" />;
    const utils = render(element);
    const input = utils.container.querySelector('input');
    const label = utils.container.querySelector('.MuiFormControlLabel-label');

    return {
      element,
      input,
      label,
      ...utils,
    };
  };

  const setup = makeSetup();

  it('It should NOT to be checked when start', () => {
    const { input } = setup;
    expect(input.checked).toBe(false);
  });

  it('It should BE checked when clicked', () => {
    const { input } = setup;
    fireEvent.change(input, { target: { checked: true } });
    expect(input.checked).toBe(true);
  });

  it('It should be correct label', () => {
    const { label } = setup;
    expect(label.textContent).toBe('Lorem Ipsum');
  });

  it('It should renders correctly', () => {
    const { element } = setup;
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
