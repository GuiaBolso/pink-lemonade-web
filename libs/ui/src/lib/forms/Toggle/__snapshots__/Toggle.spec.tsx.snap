// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Component It should renders correctly 1`] = `
<label
  className="MuiFormControlLabel-root css-1w3s9zg"
>
  <span
    className="MuiSwitch-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          className="PrivateSwitchBase-input-8 MuiSwitch-input"
          disabled={false}
          name="simple-toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb"
        />
      </span>
    </span>
    <span
      className="MuiSwitch-track"
    />
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    Lorem Ipsum
  </span>
</label>
`;
